containers/java-postgres/README.md

```diff
@@ -0,0 +1,146 @@
+# Java & PostgreSQL
+
+- [Java & PostgreSQL](#java--postgresql)
+  - [Summary](#summary)
+  - [Using this definition](#using-this-definition)
+    - [Adding another service](#adding-another-service)
+    - [Debug Configuration](#debug-configuration)
+    - [Installing Maven or Gradle](#installing-maven-or-gradle)
+    - [Installing Node.js](#installing-nodejs)
+    - [Adding the definition to your folder](#adding-the-definition-to-your-folder)
+  - [Testing the definition](#testing-the-definition)
+  - [Testing the PostgreSQL container](#testing-the-postgresql-container)
+  - [License](#license)
+
+## Summary
+
+*Develop applications with Java and PostgreSQL. Includes a Java application container and PostgreSQL server.*
+
+| Metadata | Value |  
+|----------|-------|
+| *Contributors* | The VS Code Java Team |
+| *Categories* | Core, Languages |
+| *Definition type* | Docker Compose |
+| *Available image variants* | [See Java definition](../java). |
+| *Supported architecture(s)* | x86-64, arm64/aarch64 for `bullseye` variants |
+| *Works in Codespaces* | Yes |
+| *Container host OS support* | Linux, macOS, Windows |
+| *Container OS* | Debian |
+| *Languages, platforms* | Java |
+
+## Using this definition
+
+This definition creates two containers, one for Java and one for PostgreSQL. VS Code will attach to the Java container, and from within that container the PostgreSQL container will be available on **`localhost`** port 5432. The default database is named `postgres` with a user of `postgres` whose password is `postgres`, and if desired this may be changed in `docker-compose.yml`. Data is stored in a volume named `postgres-data`.
+
+While the definition itself works unmodified, it uses the `mcr.microsoft.com/vscode/devcontainers/java` image which includes `git`, a non-root `vscode` user with `sudo` access, and a set of common dependencies and Java tools for development. You can pick a different version of this image by updating the `VARIANT` arg in `.devcontainer/docker-compose.yml` to pick a Java version.
+
+```yaml
+build:
+  context: ..
+  dockerfile: .devcontainer/Dockerfile
+    args:
+      # Update 'VARIANT' to pick an version of Java: 11, 17.
+      # Append -bullseye or -buster to pin to an OS version.
+      # Use -bullseye variants on local arm64/Apple Silicon.
+      VARIANT: 11-bullseye
+```
+
+You also can connect to PostgreSQL from an external tool when using VS Code by updating `.devcontainer/devcontainer.json` as follows:
```

Comment:
So if I'm a developer using this and wanted to populate my fancy new PostgreSQL database with test data, I'd likely do this using an external tool right? Would it be worth adding a sentence or two in here explaining how someone would go about populating some test data into their dev container instance?

Reply:
We could look into doing that. Do you think it needs to be tool specific?

Reply:
Added some guidance on populating test data into the dev container using PSQL.